### PR TITLE
Lamg: parallel solve for multiple components

### PR DESCRIPTION
**Pull request by bernlu**
This PR adds support to the Lamg solver to deal with multi-component systems in parallel.
Related issue: #1167

Todos:
- [x] add tests
- [x] make code more const correct
- [x] rework setup functions
- [x] add support for parallelSolve
- [x] benchmarks

Note that this PR depends on #1210 and #1214 which should be merged first. (currently all commits from #1210 are listed here as well, but this PR is only for changes in LAMG.hpp and tests)

### CODE REVIEW

File: include/networkit/numerics/LinearSolver.hpp [hunk 1]
```diff
@@ -134,12 +136,14 @@ void LinearSolver<Matrix>::setupConnected(const Graph &graph) {
 }
 
 template <class Matrix>
-void LinearSolver<Matrix>::parallelSolve(const std::vector<Vector> &rhs,
-                                         std::vector<Vector> &results, count maxConvergenceTime,
-                                         count maxIterations) {
+std::vector<SolverStatus>
+LinearSolver<Matrix>::parallelSolve(const std::vector<Vector> &rhs, std::vector<Vector> &results,
+                                    count maxConvergenceTime, count maxIterations) const {
+    std::vector<SolverStatus> stati(rhs.size());
```
Comment (fabratu) [inline include/networkit/numerics/LinearSolver.hpp:142]: Is this referring to the plural of `status`? Maybe then better use `stats` as a name.

File: networkit/cpp/numerics/test/LAMGGTest.cpp [hunk 2]
```diff
@@ -1,121 +1,229 @@
 /*
  * LAMGGTest.cpp
  *
- *  Created on: 20.11.2014
- *      Author: Michael
+ *  Created on: 24.04.2024
```
Comment (fabratu) [inline networkit/cpp/numerics/test/LAMGGTest.cpp:4]: Why are there now two files for testing? I think at least the file names are not self-explanatory in this case.

File: networkit/cpp/numerics/test/LAMGSolverGTest.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/*
+ * LAMGGTest.cpp
```
Comment (fabratu) [inline networkit/cpp/numerics/test/LAMGSolverGTest.cpp:2]: Wrong file name.

**Review by fabratu — CHANGES_REQUESTED** [review]

File: networkit/cpp/numerics/test/LAMGGTest.cpp (hunk 2 above)
Comment (bernlu) [reply networkit/cpp/numerics/test/LAMGGTest.cpp:4]: There used to be a `LAMGGtest` which did not actually test the `LAMG` class at all - only `SolverLamg`. I moved these tests to a new test file. I will rename it to `SolverLamgGTest` now.
For new `LAMG` tests I use the now empty `LAMGGTest` file.

**Review by fabratu — APPROVED** [review]

**Pull request merged**